### PR TITLE
DNN-20661: Make get-page testable

### PR DESCRIPTION
**Pull request by Mohtshum**
- Make get-page dependency injectable 
- Add unit test for success and failure cases

### CODE REVIEW

File: src/Modules/Content/Dnn.PersonaBar.Pages.Tests/GetPageUnitTests.cs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+            // Assert
+            Assert.IsFalse(result.IsError);
+            Assert.IsNotNull(result.Data);
+            Assert.AreEqual(1, result.Records);
+            Assert.IsFalse(result is ConsoleErrorResultModel);
+
+            _tabControllerMock.Verify(t => t.GetTab(tabId, testPortalId));
+            _securityServiceMock.Verify(s => s.CanManagePage(tabId));
+            _contentVerifierMock.Verify(c => c.IsContentExistsForRequestedPortal(testPortalId, portalSettings, false));
```
Comment (tpluscode) [inline src/Modules/Content/Dnn.PersonaBar.Pages.Tests/GetPageUnitTests.cs]: I don't think we need mock verification if they are not subject of the test

File: src/Modules/Content/Dnn.PersonaBar.Pages.Tests/GetPageUnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        {
+            // Arrange
+            var tabId = 212;
+            int testPortalId = 1;
+
+            TabInfo nullTab = null;
+
+            TabInfo tab = new TabInfo();
+            tab.TabID = tabId;
+            tab.PortalID = testPortalId;
```
Comment (tpluscode) [inline src/Modules/Content/Dnn.PersonaBar.Pages.Tests/GetPageUnitTests.cs]: `tab` is unused in this test.

File: src/Modules/Content/Dnn.PersonaBar.Pages.Tests/GetPageUnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+            PortalSettings portalSettings = new PortalSettings();
+            portalSettings.PortalId = testPortalId;
+
+            _tabControllerMock.SetReturnsDefault(nullTab);
+            _securityServiceMock.SetReturnsDefault(true);
+            _contentVerifierMock.SetReturnsDefault(false);
```
Comment (tpluscode) [inline src/Modules/Content/Dnn.PersonaBar.Pages.Tests/GetPageUnitTests.cs]: Common default behaviour can be placed in the `[SetUp]` method to declutter individual tests.

**Review by tpluscode — CHANGES_REQUESTED** [review]

File: src/Modules/Content/Dnn.PersonaBar.Pages.Tests/GetPageUnitTests.cs (hunk 3 above)
Comment (Mohtshum) [reply src/Modules/Content/Dnn.PersonaBar.Pages.Tests/GetPageUnitTests.cs]: Not exactly same for each test

File: src/Modules/Content/Dnn.PersonaBar.Pages.Tests/GetPageUnitTests.cs (hunk 1 above)
Comment (Mohtshum) [reply src/Modules/Content/Dnn.PersonaBar.Pages.Tests/GetPageUnitTests.cs]: Mock verification helping the logical flow for test, as if tab null in that case bypass security and content verification. I think it should stay any way I can remove them if you want :(

File: src/Modules/Content/Dnn.PersonaBar.Pages.Tests/GetPageUnitTests.cs (hunk 3 above)
Comment (Mohtshum) [reply src/Modules/Content/Dnn.PersonaBar.Pages.Tests/GetPageUnitTests.cs]: While I think you mean setting up of initializing tab, portalSettings and getComands

**Review by tpluscode — APPROVED** [review]

**Pull request merged**